rita_common/src/payment_validator/mod.rs

```diff
@@ -74,14 +75,18 @@ impl fmt::Display for ToValidate {
 
 pub struct PaymentValidator {
     unvalidated_transactions: HashSet<ToValidate>,
-    successful_transactions: HashSet<Uint256>,
+    /// All successful transactions sent FROM this router, mapped To Address-> list of PaymentTx
+    successful_transactions_sent: HashMap<Identity, HashSet<PaymentTx>>,
+    /// All successful transactions sent TO this router, mapped From Address-> list of PaymentTx
+    successful_transactions_rec: HashMap<Identity, HashSet<PaymentTx>>,
```

Comment:
I think all we need to store here is txid's, what about the rest of the tx do we use?

Reply:
If the client sends a list of old pmts over to exit, and the exit sees an old pmt that needs to be accounted for, doesnt the exit need the whole pmt struct to pass it to validate_later?

Reply:
Fixed